sklearn/feature_selection/tests/test_from_model.py

```diff
@@ -41,3 +48,83 @@ def test_invalid_input():
+    transformer = SelectFromModel(estimator=est)
+    transformer.fit(iris.data, iris.target)
+    assert_equal(transformer.estimator, est)
+
+
+def test_feature_importances():
+    X, y = datasets.make_classification(n_samples=1000,
+                                        n_features=10,
+                                        n_informative=3,
+                                        n_redundant=0,
+                                        n_repeated=0,
+                                        shuffle=False,
+                                        random_state=0)
+
+    est = RandomForestClassifier(n_estimators=50, random_state=0)
+    transformer = SelectFromModel(estimator=est)
+
+    transformer.fit(X, y)
+    assert_true(hasattr(transformer.estimator_, 'feature_importances_'))
+
+    X_new = transformer.transform(X)
+    assert_less(X_new.shape[1], X.shape[1])
+
+    feature_mask = (transformer.estimator_.feature_importances_ >
+                    transformer.estimator_.feature_importances_.mean())
+    assert_array_almost_equal(X_new, X[:, feature_mask])
+
+    # Check with sample weights
+    sample_weight = np.ones(y.shape)
+    sample_weight[y == 1] *= 100
+
+    est = RandomForestClassifier(n_estimators=50, random_state=0)
+    transformer = SelectFromModel(estimator=est)
+    transformer.fit(X, y, sample_weight=sample_weight)
+    importances = transformer.estimator_.feature_importances_
+    assert_less(importances[1], X.shape[1])
+
+    est = RandomForestClassifier(n_estimators=50, random_state=0)
+    transformer = SelectFromModel(estimator=est)
+    transformer.fit(X, y, sample_weight=3*sample_weight)
+    importances_bis = transformer.estimator_.feature_importances_
+    assert_almost_equal(importances, importances_bis)
+
+
+def test_partial_fit():
+    est = PassiveAggressiveClassifier()
+    transformer = SelectFromModel(estimator=est)
+    transformer.partial_fit(iris.data, iris.target,
+                            classes=np.unique(iris.target))
+    id_1 = id(transformer.estimator_)
+    transformer.partial_fit(iris.data, iris.target,
+                            classes=np.unique(iris.target))
+    id_2 = id(transformer.estimator_)
+    assert_equal(id_1, id_2)
+
+
+def test_warm_start():
+    est = PassiveAggressiveClassifier(warm_start=True)
+    transformer = SelectFromModel(estimator=est)
+    transformer.fit(iris.data, iris.target)
+    id_1 = id(transformer.estimator_)
+    transformer.fit(iris.data, iris.target)
+    id_2 = id(transformer.estimator_)
+    assert_equal(id_1, id_2)
+
+
+def test_fitted_estimator():
+    """Test that a fitted estimator can be passed to SelectFromModel."""
+    clf = SGDClassifier(alpha=0.1, n_iter=10, shuffle=True, random_state=0)
+    model = SelectFromModel(clf)
+    model.fit(iris.data, iris.target)
+    X_transform = model.transform(iris.data)
+
+    clf.fit(iris.data, iris.target)
+    model = SelectFromModel(clf)
+    assert_array_equal(model.transform(iris.data), X_transform)
```

Comment:
I'm not sure exactly what @glouppe is referring to, whether he means strings, etc. But I'd like to see a test that what you do in your Boston example will work: modifying the threshold without re-fitting.